src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
+
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function averageGrades(grades: number[]): number {
+  return grades
+    .reduce((sum:number, grade: number) => (sum + grade), 0) / grades.length;
+}
 
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
   // write your function
+  const copyStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyStudents.sort(
+        (a: Student, b: Student) => (order === 'asc'
```

Comment:
TS knows about this types here
```suggestion
        (a, b) => (order === 'asc'
```